gemrb/plugins/SDLVideo/SDLSurfaceSprite2D.cpp

```diff
@@ -119,27 +119,25 @@ Palette* SDLSurfaceSprite2D::GetPalette() const
 		const Color* end = begin + fmt->palette->ncolors;
 		palette = new Palette(begin, end);
 	}
-	palette->acquire();
 	surface->palette = palette;
 
 	return palette;
 }
 
 bool SDLSurfaceSprite2D::IsPaletteStale() const
 {
-	Palette* pal = GetPalette();
+	PaletteHolder pal = GetPalette();
 	return pal && pal->GetVersion() != palVersion;
 }
 
-void SDLSurfaceSprite2D::SetPalette(Palette* pal)
+void SDLSurfaceSprite2D::SetPalette(PaletteHolder pal)
 {
-	Palette* palette = surface->palette.get();
+	PaletteHolder palette = surface->palette.get();
 
 	// we don't use shared palettes because it is a performance bottleneck on SDL2
-	assert(pal != palette);
+	assert(pal.get() != palette.get());
```

Comment:
we should have `operator==` overloads defined for this

```suggestion
	assert(pal != palette);
```

Reply:
I actually tried that first, and changed it when it errored out. The overload for operator!= of two Holders is missing, I'll add it to the patch.